### PR TITLE
Fix issues with restoring snapshot during onboarding

### LINKED ISSUE #9009: Snapshot loading wrong image during on boarding restore
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

I have been doing numerous restores due to a issue I am having with my Rpi4 since November 5.5 update.  I each restore I go to the screen where you either create a new user or load the snapshot during on boarding.  I see a spinner while the snapshot is downloading to the Pi.  I then get to the screen asking me what I want to restore (all, partial, etc.)  When I click the all I get the previous screen reloaded asking me to create a new user or restore a snapshot rather than a new screen showing it is being restored.

### Describe the behavior you expected

I would expect a new screen either showing that the snapshot is being restored or ideally a screen showing the progress of the restore.

### Steps to reproduce the issue

1.  Install an image on a pi 4. (Tried each version since 5.8)
2.  Boot the Pi and wait for the new user screen to appear in Chrome ( running the latest versions at the time of HA releases).  There have been updates since I first noticed this issue.
3.  Click on restore image.
...

### What version of Home Assistant Core has the issue?

Core-2021.4.6

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Google Chrome Version 90.0.4430.93 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

Windows Home 10.  experience pack 120.2212.551.0

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

No errors are showing up anywhere that I have seen.

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #9009
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/onboarding/ha-onboarding.ts [hunk 1]
```diff
@@ -87,6 +92,7 @@ class HaOnboarding extends litLocalizeLiteMixin(HassElement) {
           ? html`<onboarding-restore-snapshot
               .localize=${this.localize}
               .restoring=${this._restoring}
+              ._discoveryInformation=${this._discoveryInformation}
```
Comment (bramkragten) [inline src/onboarding/ha-onboarding.ts:95]:
```suggestion
              .discoveryInformation=${this._discoveryInformation}
```

File: src/onboarding/onboarding-restore-snapshot.ts [hunk 2]
```diff
@@ -58,11 +65,13 @@ class OnboardingRestoreSnapshot extends ProvideHassLitMixin(LitElement) {
   private async _checkRestoreStatus(): Promise<void> {
     if (this.restoring) {
       try {
-        const response = await fetch("/api/hassio/supervisor/info", {
-          method: "GET",
-        });
-        if (response.status === 401) {
-          // If we get a unauthorized response, the restore is done
+        const response = await fetchDiscoveryInformation();
```
Comment (bramkragten) [inline src/onboarding/onboarding-restore-snapshot.ts:68]: We no longer need to fetch it here now?

Comment (ludeeus) [reply src/onboarding/onboarding-restore-snapshot.ts:68]: yes we do

File: src/onboarding/onboarding-restore-snapshot.ts [hunk 3]
```diff
@@ -58,11 +65,13 @@ class OnboardingRestoreSnapshot extends ProvideHassLitMixin(LitElement) {
...
+
+        if (
+          !this.discoveryInformation ||
+          this.discoveryInformation.uuid !== response.uuid
+        ) {
+          // When the UUID changes, the restore is complete
           navigate("/", { replace: true });
           location.reload();
```
Comment (bramkragten) [inline src/onboarding/onboarding-restore-snapshot.ts:76]: Why use `navigate` if we are going to reload anyway? Can't we use `window.location.replace`?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**